eternalcore-core/src/main/java/com/eternalcode/core/feature/jail/JailServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.eternalcode.core.feature.jail;
+
+import com.eternalcode.commons.bukkit.position.Position;
+import com.eternalcode.commons.bukkit.position.PositionAdapter;
+import com.eternalcode.core.configuration.ConfigurationManager;
+import com.eternalcode.core.configuration.implementation.LocationsConfiguration;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.injector.annotations.component.Service;
+import org.bukkit.Location;
+
+import java.util.Optional;
+
+@Service
+public class JailServiceImpl implements JailService {
+
+    private final LocationsConfiguration locationsConfiguration;
+    private final ConfigurationManager configurationManager;
+
+    @Inject
+    public JailServiceImpl(LocationsConfiguration locationsConfiguration, ConfigurationManager configurationManager) {
+        this.locationsConfiguration = locationsConfiguration;
+        this.configurationManager = configurationManager;
+    }
+
+    @Override
+    public Optional<Location> getJailLocation() {
+        return Optional.of(PositionAdapter.convert(this.locationsConfiguration.jail));
+    }
+
+    @Override
+    public void setupJailArea(Location jailLocation) {
+        this.locationsConfiguration.jail = PositionAdapter.convert(jailLocation);
+        this.configurationManager.save(this.locationsConfiguration);
```

Comment:
This is IO operation. Add @Blocking annotation (and also add it to the same method in interface) and enable async processing in jail setup command by using @Async before declaration of this command.